test/src/e2e_vm_tests/test_programs/should_fail/array_wrong_elements_types/stdout.snap

```diff
@@ -7,6 +7,43 @@ output:
     Building test/src/e2e_vm_tests/test_programs/should_fail/array_wrong_elements_types
    Compiling library core (sway-lib-core)
    Compiling library std (sway-lib-std)
+warning
+   --> sway-lib-std/src/ecr.sw:221:26
+    |
+219 | #[deprecated(note = "std::ecr has been replaced by std::crypto, and is no longer maintained")]
+220 | pub fn ec_recover_address(signature: B512, msg_hash: b256) -> Result<Address, EcRecoverError> {
+221 |     let pub_key_result = ec_recover(signature, msg_hash);
```

Comment:
This means that `std` compiles with warnings. And it is essentially a deprecated function calling another deprecated function from the same module.

Adding `#[allow(deprecated)]` on all those functions next to the `#[deprecated]` will remove this issue. They will still be deprecated for all other usages, but will allow calling other deprecated functions from the same module.

Reply:
I didn't dig into it but if those are indeed deprecated functions calling each other I agree.